tests/test_project/admin.py

```diff
@@ -1,12 +1,35 @@
 from django.contrib import admin
-from openwisp_utils.admin import ReadOnlyAdmin
+from django.contrib.admin import ModelAdmin
+from django.contrib.admin.options import InlineModelAdmin
```

Comment:
this is wrong, you shouldn't use this class, where did you read this? You have to use `model.StackedInline`, I will explain further in the next comments